lib/Graph/Nodes.cpp

```diff
@@ -862,6 +862,32 @@ bool GatherNode::verify() const {
   return isValid;
 }
 
+bool GatherRangesNode::verify() const {
+  bool isValid = expectCompareTrue(
```

Comment:
it would be good to reject op earlier in the pipeline, at the moment of actual loading in the loader.